packages/slice-machine/src/features/onboarding/helpers/useOnboardingProgress.ts

```diff
@@ -0,0 +1,70 @@
+import { Dispatch, SetStateAction, useEffect, useRef } from "react";
+
+import { onboardingSteps as steps } from "@/features/onboarding/content";
+import type {
+  OnboardingStep,
+  OnboardingStepStatuses,
+  OnboardingStepType,
+} from "@/features/onboarding/types";
+import { useLocalStorageItem } from "@/hooks/useLocalStorageItem";
+
+const getInitialState = (): OnboardingStepStatuses => {
+  // if the old guide was dismissed, all steps start as complete
+  const startComplete =
+    localStorage.getItem("slice-machine_isInAppGuideOpen") === "false";
+
+  return Object.fromEntries(
+    steps.map((step) => [step.id, startComplete]),
+  ) as OnboardingStepStatuses;
+};
+
+const useOnboardingStepStatus = (): [
+  OnboardingStepStatuses,
+  Dispatch<SetStateAction<OnboardingStepStatuses>>,
+] => {
+  const initialState = useRef(getInitialState()).current;
```

Comment:
Forgot this one, it's not recommended to read (or set) refs during the render of a component: https://react.dev/reference/react/useRef#caveats

Reply:
That's a nice catch, thanks 🙂